src/pages/index.js

```diff
@@ -39,34 +39,90 @@ function createCard(cardData) {
 function renderCard(cardData) {
   const card = new Card(cardData, "#card-template", {
     handleImageClick: () => imagePopup.open(cardData),
-  });
+  }, handleDeleteCardSubmit, toggleLikeCard);
   return card.getView();
 }
 
-// HANDLERS
-function handleFormSubmit(data) {
-  userInfo.setUserInfo(data);
-  profileEditPopup.close();
+function toggleLikeCard(card) {
+    if(!card._isLiked) {
+      api.addLikeCard(card._id)
+      .then(() => {
+        card.toggleLikeButton();
+      })
+    } else {
+      api.removeLikeCard(card._id)
+      .then(() => {
+        card.toggleLikeButton();
+      });
+      }
+}
+
+// ~~~~~~~~~~~~~~~ HANDLERS ~~~~~~~~~~~~~~ //
+// -------- PROFILE ------- //
+function handleProfileEditSubmit(data) {
+  profileEditPopup.isLoading(true, "Saving...");
+  api.editUserInfo(data)
+  .then((userData) => {
+    newProfileInfo.setUserInfo(userData);
+    profileEditPopup.close();
+  })
+  .catch((err) => {
+    console.error(err);
+  })
+  .finally(() => profileEditPopup.isLoading(false));
 }
 
+function handleProfileImageSubmit(data) {
+  profileImageEditPopup.isLoading(true, "Saving...");
+  api.editProfileImage(data.url)
+  .then((res) => {
+    newProfileInfo.setProfileImage(res.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  })
+  .finally(() => {
+    profileImageEditPopup.isLoading(false);
+    profileImageEditPopup.close();
+  })
+}
+
+// --------- CARD --------- //
 function handleAddCardSubmit(data) {
-  const name = data.title;
-  const link = data.url;
-  createCard({name, link});
-  cardPopup.close();
+  cardPopup.isLoading(true, "Saving...");
+  api.addCard(data)
+  .then((newCard) => {
+    createCard(newCard);
+  })
+  .catch((err) => {
+    console.error(err)
+  })
+  .finally(() => {
+    cardPopup.close()
+  cardPopup.isLoading(false);
+  })
 }
 
-const cardSection = new Section(
-  { items: initialCards, renderer: createCard },
-  cardListEl
-);
-cardSection.renderItems();
+function handleDeleteCardSubmit(card) {
+  cardDeletePopup.open();
+  cardDeletePopup.setSubmitCallback(() => {
+    cardDeletePopup.isLoading(true, "Saving...");
+    api.deleteCard(card)
+    .then(() => {
+      card.deleteCard(), cardDeletePopup.close();
```

Comment:
Needs correction: don't use inline statements, since it reduces the code readability:

```js
card.deleteCard();
cardDeletePopup.close();
```